man/systemd-system.conf.xml

```diff
@@ -162,6 +162,30 @@
         These settings have no effect if a hardware watchdog is not available.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>PreTimeoutWatchdogSec=</varname></term>
+
+        <listitem><para>Configure the hardware watchdog device pretimeout
+        value. Takes a timeout value in seconds (or in other time units
+        similar to <varname>RuntimeWatchdogSec=</varname>). A watchdog
+        pretimeout is a notification generated by the watchdog before the
+        watchdog reset might occur in the event the watchdog has not been
+        serviced. This notification is handled by the kernel and can be
+        configured to take an action (i.e. generate a kernel panic). Not
+        all watchdog hardware or drivers support generating a pretimeout
+        and depending on the state of the system, the kernel may be unable
+        to take the configured action before the watchdog reboot. The
+        watchdog will be configured to generate the pretimeout event at
+        the amount of time specified by
+        <varname>PreTimeoutWatchdogSec=</varname> before the runtime
+        watchdog timeout (set by <varname>RuntimeWatchdogSec=</varname>).
+        By default, <varname>PreTimeoutWatchdogSec=</varname> defaults to
+        0 (off). This setting has no effect if a hardware watchdog is not
+        available or the hardware watchdog does not support a pretimeout
+        and will be ignored by the kernel if the setting is greater than
+        the actual watchdog timeout.</para></listitem>
+      </varlistentry>
```

Comment:
this should probably say that this timeout runs parallel to RuntimeWatchdogSec=, i.e. is also reset whenever the hw watchdog is pinged, and thus should always be chosen smaller than RuntimeWatchdogSec=. (that's how i understood it?)

Reply:
I'm not sure "parallel" is the right word, since the pretimeout determines the time _before_ the timeout that it fires.

So for example with `RuntimeWatchdogPreSec=10` and `RuntimeWatchdogSec=60`, the pretimeout would fire at 50s and the watchdog reset would fire at 60s.

This is just how the kernel uses the value. Does it make sense to flip the meaning so that `RuntimeWatchdogPreSec=10` means the pretimeout would fire after 10s (regardless of what `RuntimeWatchdogSec` is set to)? This might be simpler to understand and would be consistent with how we use the other `*WatchdogSec` settings.

Reply:
Ah, I had incorrectly assumed a value of 10s for the pre-timeout meant a pre-timeout event would occur at 10s regardless of what the normal timeout is set to.

I think we should keep things as you describe (pre-timeout of 10s means the event will happen 10s before the normal timeout) and just be sure to document it well. I think deviating from how the kernel interprets the value is probably not the right way to go.